frontend/packages/console-shared/src/utils/keyword-filter.ts

```diff
@@ -0,0 +1,19 @@
+export const keywordFilter = <T>(
+  queryString: string,
+  items: T[],
+  compareFunc: (keyword: string, item: T) => boolean,
+): T[] => {
+  if (!queryString) {
+    return items;
+  }
+  const keywords = queryString.trim().split(' ');
+
+  return items.filter((item) => {
+    return keywords.every((keyword) => {
+      if (!keyword) {
+        return false;
+      }
+      return compareFunc(keyword.toLowerCase(), item);
```

Comment:
Don't you need to lowercase item, too?

It would be better to lowercase the keyword once when creating the array instead of on every comparison.

Reply:
Since the item could be of any type, I've moved the lowercasing to `compareFunction`.